streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -831,23 +867,60 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
 
     /**
      * Get the configs for the {@link KafkaConsumer restore-consumer}.
-     * Properties using the prefix {@link #CONSUMER_PREFIX} will be used in favor over their non-prefixed versions
+     * Properties using the prefix {@link #RESTORE_CONSUMER_PREFIX} will be used in favor over their non-prefixed versions
      * except in the case of {@link ConsumerConfig#BOOTSTRAP_SERVERS_CONFIG} where we always use the non-prefixed
      * version as we only support reading/writing from/to the same Kafka Cluster.
+     * If not specified by {@link #RESTORE_CONSUMER_PREFIX}, restore consumer will share the general consumer configs
+     * prefixed by {@link #CONSUMER_PREFIX}.
```

Comment:
This sounds reasonable for backward compatibility -- however, it make the "overwriting" logic/hierarchy quite complex. I am wondering, if we might want to break compatibility to keep the logic simpler... I'll comment on the KIP thread to discuss this. (no need to reply here)